src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Aapt2.targets

```diff
@@ -0,0 +1,173 @@
+<!--
+***********************************************************************************************
+Xamarin.Android.Aapt2.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+  created a backup copy.  Incorrect changes to this file will make it
+  impossible to load or build your projects from the command-line or the IDE.
+
+This file imports the version- and platform-specific targets for the project importing
+this file. This file also defines targets to produce an error if the specified targets
+file does not exist, but the project is built anyway (command-line or IDE build).
+
+Copyright (C) 2010-2011 Novell. All rights reserved.
+Copyright (C) 2011-2012 Xamarin. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+<Target Name="_ReadAapt2VersionCache">
+  <ReadLinesFromFile File="$(_AndroidAapt2VersionFile)"
+      Condition="Exists('$(_AndroidAapt2VersionFile)')">
+    <Output TaskParameter="Lines" ItemName="_Aapt2VersionCache"/>
+  </ReadLinesFromFile>
+</Target>
+
+<Target Name="_CreateAapt2VersionCache"
+    DependsOnTargets="_ReadAapt2VersionCache"
+    AfterTargets="_SetLatestTargetFrameworkVersion"
+    Condition="'$(_AndroidUseAapt2)' == 'True' And '$(_Aapt2Version)' != '@(_Aapt2VersionCache)'"
+  >
+  <MakeDir Directories="$(IntermediateOutputPath)" Condition="!Exists('$(IntermediateOutputPath)')" />
+  <WriteLinesToFile
+      Condition="'$(_Aapt2Version)' != '@(_Aapt2VersionCache)'" 
+      File="$(_AndroidAapt2VersionFile)"
+      Lines="$(_Aapt2Version)"
+      Overwrite="true"
+  />
+  <ItemGroup Condition="'$(_Aapt2Version)' != '@(_Aapt2VersionCache)'">
+      <_CompiledFlataArchive Include="$(_AndroidLibrayProjectIntermediatePath)**\*.flata" />
```

Comment:
Xml indentation needs fixing at various points, but we can do that before the final merge.